commit 12979f4c3d345759c39d74b946a3caf924af95a3
Author: Dave Stevenson <dave.stevenson@raspberrypi.org>
Date:   Wed Feb 13 12:36:56 2019 +0000

    staging: mmal-vchiq: Always return the param size from param_get
    
    mmal-vchiq is a reimplementation of the userland library for MMAL.
    When getting a parameter, the client provides the storage and
    the size of the storage. The VPU then returns the size of the
    parameter that it wished to return, and as much as possible of
    that parameter is returned to the client.
    
    The implementation previously only returned the size provided
    by the VPU should it exceed the buffer size. So for parameters
    such as the supported encodings list the client had no idea
    how much of the provided storage had been populated.
    
    Signed-off-by: Dave Stevenson <dave.stevenson@raspberrypi.org>

diff --git a/drivers/staging/vc04_services/vchiq-mmal/mmal-vchiq.c b/drivers/staging/vc04_services/vchiq-mmal/mmal-vchiq.c
index 167644137ec7..3ef173f2d18b 100644
--- a/drivers/staging/vc04_services/vchiq-mmal/mmal-vchiq.c
+++ b/drivers/staging/vc04_services/vchiq-mmal/mmal-vchiq.c
@@ -1412,11 +1412,12 @@ static int port_parameter_get(struct vchiq_mmal_instance *instance,
 		 */
 		memcpy(value, &rmsg->u.port_parameter_get_reply.value,
 		       *value_size);
-		*value_size = rmsg->u.port_parameter_get_reply.size;
 	} else {
 		memcpy(value, &rmsg->u.port_parameter_get_reply.value,
 		       rmsg->u.port_parameter_get_reply.size);
 	}
+	/* Always report the size of the returned parameter to the caller */
+	*value_size = rmsg->u.port_parameter_get_reply.size;
 
 	pr_debug("%s:result:%d component:0x%x port:%d parameter:%d\n", __func__,
 		 ret, port->component->handle, port->handle, parameter_id);
diff -uN a/1.txt b/1.txt
--- a/dummy/rpi_1358_12979f4c3d345759c39d74b946a3caf924af95a3.txt	1970-01-01 00:00:00.000000000 +0000
+++ b/dummy/rpi_1358_12979f4c3d345759c39d74b946a3caf924af95a3.txt	2013-12-23 04:07:40.000000000 +0000
@@ -0,0 +1 @@
+dummy file to ensure patch has content.
